### PR TITLE
Tappx: Remove Host Parameter

### PR DESCRIPTION
**Pull request by prebidtappx**
In this pull request we solve the hostname problem (https://github.com/prebid/prebid-server/issues/2127). The solution has been to remove the host parameter and try to adapt everything to prebid's needs. However, we have made some of the code backward compatible.
In conclusion, we have made the changes to prevent prebid from being used as a proxy.

### CODE REVIEW

**Comment by bretg** [comment]
Thanks for the action on this @prebidtappx

**Comment by prebidtappx** [comment]
Hi! All the changes are done! We need to do something else or is in review progress?
Thanks for everything!

**Comment by mansinahar** [comment]
@prebidtappx This is on our list to take a look at today. Thanks!

File: adapters/tappx/tappx.go [hunk 1]
```diff
@@ -5,8 +5,9 @@ import (
 	"fmt"
 	"net/http"
 	"net/url"
+	"regexp"
 	"strconv"
-	"strings"
+	//"strings"
```
Comment (mansinahar) [inline adapters/tappx/tappx.go:10]: Can you please remove this import if it is no longer needed?

**Review by mansinahar — COMMENTED** [review]
@prebidtappx LGTM generally except the very minor comment above and also, can you please open up a docs PR for this as well over here: https://github.com/prebid/prebid.github.io/blob/master/dev-docs/bidders/tappx.md ? Thanks!

File: adapters/tappx/tappx.go (hunk 1 above)
Comment (prebidtappx) [reply adapters/tappx/tappx.go]: Removed the import in commit 793b2b4
Updated the docs in pull request: https://github.com/prebid/prebid.github.io/pull/3520

File: adapters/tappx/tappx.go [hunk 2]
```diff
@@ -137,7 +131,15 @@ func (a *TappxAdapter) buildEndpointURL(params *openrtb_ext.ExtImpTappx, test in
 		}
 	}
 
-	endpointParams := macros.EndpointTemplateParams{Host: params.Host}
+	tappxHost := "tappx.com"
+	isNewEndpoint, err := regexp.Match(`^(zz|vz)[0-9]{3,}([a-z]{2}|test)$`, []byte(params.Endpoint))
+	if isNewEndpoint {
+		tappxHost = params.Endpoint + ".pub." + tappxHost + "/rtb/"
```
Comment (mansinahar) [inline adapters/tappx/tappx.go:137]: @prebidtappx Quick question: Is the `params.Endpoint + ".pub." + tappxHost + "/rtb/"` going to be the new Tappx endpoint and is the plan to deprecate `tappxHost = "ssp.api." + tappxHost + "/rtb/v2/"` in the future?

Comment (prebidtappx) [reply adapters/tappx/tappx.go:137]: Yes, it is the plan. However, a lot of people don't like the changes. For this reason, all the changes are backward compatible.

**Review by bretg — COMMENTED** [review]
(sorry, meant to put this in the docs PR)
appreciate the doc update @prebidtappx , but since you support both Prebid Server and Prebid.js, please confirm that the host param is optional for both. I don't see any PRs for PBJS.

**Review by hhhjort — APPROVED** [review]
LGTM

**Review by mansinahar — APPROVED** [review]

**Pull request merged**